lightbeam/test.py

```diff
@@ -0,0 +1,214 @@
+        
+        # * Test `lightbeam truncate`:
+        self.logger.info("testing `lighbeam truncate`...")
+        pre_truncate_count_file = os.path.join( tests_dir, "output", "truncate-pre-count.tsv" )
+        self.lightbeam.results_file = pre_truncate_count_file
+        self.lightbeam.selector = "disciplineActions"
+        self.lightbeam.api.prepare()
+        self.lightbeam.counter.count()
+        pre_count = self.load_count_file(pre_truncate_count_file)
+        
+        self.lightbeam.results_file = ""
+        self.lightbeam.selector = "disciplineActions"
+        self.lightbeam.drop_keys = "id,_etag,_lastModified"
+        self.lightbeam.api.prepare()
+        self.lightbeam.fetcher.fetch()
+        records_jsonl_file = os.path.join( tests_dir, "data", "disciplineActions.jsonl" )
+        num_records = self.num_nonempty_lines_in_jsonl_file(records_jsonl_file)
+        # Just to be super safe, make sure we've fetched the same number of records as we counted:
+        if num_records != pre_count.get("disciplineActions"):
+            self.logger.critical(f"Preparation for `lightbeam truncate` resulted in a discrepancy between `count` and `fetch`. (No data has been deleted.)")
+        
+        self.lightbeam.results_file = ""
+        self.lightbeam.selector = "disciplineActions"
+        self.lightbeam.drop_keys = ""
+        self.lightbeam.api.prepare()
+        self.lightbeam.truncator.truncate()
+
+        post_truncate_count_file = os.path.join( tests_dir, "output", "truncate-post-count.tsv" )
+        self.lightbeam.results_file = post_truncate_count_file
+        self.lightbeam.selector = "disciplineActions"
+        self.lightbeam.api.prepare()
+        self.lightbeam.counter.count()
+        post_count = self.load_count_file(post_truncate_count_file)
+        if post_count.get("disciplineActions") != 0:
+            self.logger.critical(f"`lightbeam truncate` appears did not actually truncate the resource.")
+        
+        self.lightbeam.results_file = ""
+        self.lightbeam.selector = "disciplineActions"
+        self.lightbeam.api.prepare()
+        self.lightbeam.sender.send()
+
+        post_truncate_send_count_file = os.path.join( tests_dir, "output", "truncate-post-send-count.tsv" )
+        self.lightbeam.results_file = post_truncate_send_count_file
+        self.lightbeam.selector = "disciplineActions"
+        self.lightbeam.api.prepare()
+        self.lightbeam.counter.count()
+        if not self.count_files_are_identical(pre_truncate_count_file, post_truncate_send_count_file):
+            self.logger.critical("`lightbeam send` did not restore the original records after `truncate`")
+        
+        os.remove(records_jsonl_file)
+        output_dir = os.path.join( tests_dir, "output")
+        visible_files = [file for file in os.listdir(output_dir) if not file.startswith('.')]
+        for file in visible_files:
+            os.remove( os.path.join( output_dir, file) )
+        
+    
```

Comment:
The below are helper functions used by the test suite. An argument could be made to move some of them into `lightbeam/util.py` but they seem fine here for now.